<launch>
  <!-- Load the URDF into the ROS Parameter Server -->
  <param name="robot_description" command="$(find xacro)/xacro --inorder '$(find vawt_1)/urdf/vawt_1.xacro'" />

    <rosparam file="$(find vawt_hardware_interface)/config/hardware.yaml" command="load"/>
    <rosparam file="$(find vawt_hardware_interface)/config/controllers.yaml" command="load"/>
    <rosparam file="$(find vawt_hardware_interface)/config/joint_limits.yaml" command="load"/>
    <node name="vawt_hardware_interface" pkg="vawt_hardware_interface" type="vawt_hardware_interface_node" output="screen"/>
    <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen"
        args="/vawt_1/controller/state
        /vawt_1/controller/position/blade_1_position_controller
        /vawt_1/controller/position/blade_2_position_controller
        /vawt_1/controller/position/main_shaft_joint_position_controller
        "/>
  <!-- convert joint states to TF transforms for rviz, etc -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"
    respawn="false" output="screen">
    <remap from="/joint_states" to="/vawt_1/joint_states" />
  </node>
</launch>